### PR TITLE
json reporter: Add network identifiers

### PR DESCRIPTION
**Pull request by kevin-j-smith**
inspec/train enhanced to add hostname, domainname, fqdn and network to the platform section of it's os export.  Added logic to properly handle and report these findings.

#3869

### CODE REVIEW

**Comment by kevin-j-smith** [comment]
This commit will be more impactful once inspec/train has accepted PR: inspec/train#418 https://github.com/inspec/train/pull/418

File: lib/inspec/formatters/base.rb [hunk 1]
```diff
@@ -68,6 +68,11 @@ def stop(notification)
       run_data[:platform] = {
         name: platform(:name),
         release: platform(:release),
+        network: platform(:network),
```
Comment (clintoncwolfe) [inline lib/inspec/formatters/base.rb:71]: As commented on #3869, I really think this data should come from something like an Ohai resource (see #3886). You can access resources even in places like this by using the Resource Registry.

File: lib/resources/platform.rb [hunk 2]
```diff
@@ -19,7 +19,7 @@ def initialize
     end
 
     # add helper methods for easy access of properties
-    %w{family release arch}.each do |property|
+    %w{family release arch network hostname fqdn domain_name guid}.each do |property|
```
Comment (clintoncwolfe) [inline lib/resources/platform.rb:22]: As mentioned elsewhere, I don't think this should be added to platform.

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]
Thanks for the contribution!

However, as discussed on #3869, we'd rather see the network identifiers collected by something an Ohai resource, or something else - not from platform.

Additionally, in its final form, we generally expect tests and documentation to be included with PRs.  Thanks!

**Comment by clintoncwolfe** [comment]
(boilerplate message)
Hi! This PR has been untouched for more than two weeks, in the "Changes Requested" state. Please either make the requested changes, or continue the discussion in comments.

**Comment by kekaichinose** [comment]
__What’s happening? Why was this issue closed?__
This issue was closed due to some much needed review of legacy issues or issues that were spawned in older versions of InSpec, i.e. < v3.

__Why do I care?__
You would care about this if this was an issue that you are still seeing and/or feel needs to be addressed in the current version of InSpec.

__What do I need to do?__
If this issue is no longer important, no further action is necessary. However, if you think this is something that should be addressed, please open a new issue and refer to the original issue in the description.

**Comment by kekaichinose** [comment]
Sorry, prematurely pulled the trigger on this. Reopening.

**Comment by clintoncwolfe** [comment]
We're going to go ahead and close this, as it now has merge conflicts, hasn't been updated, and we've updated the originating issue with next steps. Thanks!

**Pull request closed without merging**